### PR TITLE
Use operator specific messaging in division by zero diagnostics

### PR DESCRIPTION
**Pull request by zanieb**
Requested at https://github.com/astral-sh/ruff/pull/13576#discussion_r1782530971

### CODE REVIEW

File: crates/red_knot_python_semantic/src/types/infer.rs [hunk 1]
```diff
@@ -4216,8 +4223,8 @@ mod tests {
             "src/a.py",
             &[
                 "Cannot divide object of type 'Literal[1]' by zero.",
-                "Cannot divide object of type 'Literal[2]' by zero.",
-                "Cannot divide object of type 'Literal[3]' by zero.",
+                "Cannot floor divide object of type 'Literal[2]' by zero.",
+                "Cannot reduce object of type 'Literal[3]' modulo zero.",
```
Comment (zanieb) [inline crates/red_knot_python_semantic/src/types/infer.rs:4227]: This is the most consistent language I can find for this online.

Comment (AlexWaygood) [reply crates/red_knot_python_semantic/src/types/infer.rs:4227]: could possibly do something like this, but it's annoyingly verbose:

```suggestion
                "Using modulo with a zero divisor is undefined on an object of type 'Literal[3]'.",
```

**Review by AlexWaygood — APPROVED** [review]
Thank you!!

File: crates/red_knot_python_semantic/src/types/infer.rs (hunk 1 above)
Comment (zanieb) [reply crates/red_knot_python_semantic/src/types/infer.rs:4227]: Yeah that seems too verbose to me and differs quite a bit from the other messages.

Comment (AlexWaygood) [reply crates/red_knot_python_semantic/src/types/infer.rs:4227]: I think what you have now is fine!

**Pull request merged**